sycl/test/on-device/sub_group_algorithms/all_of.cpp

```diff
@@ -0,0 +1,72 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -I . -o %t.out
+// XUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include "support.h"
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <cassert>
+#include <numeric>
+using namespace sycl;
+
+template <class Predicate> class all_of_kernel;
+
+struct GeZero {
```

Comment:
Do we really need thee different comparators to test the feature? I think one should be enough and save CI machine resources.